lib/vimaly/client.rb

```diff
@@ -54,13 +54,37 @@ def update_ticket(id, other_fields)
       end
     end
 
-    def search_tickets(query_string, state: 'active')
+    def search_tickets(query_string, state: 'active', max_results: 500)
       ticket_type_map = Hash[ticket_types.map { |t| [t.id, t] }]
       bin_map = Hash[bins.map { |b| [b.id, b] }]
 
-      get("/ticket-search?text=#{query_string}&state=#{state}").map do |ticket_data|
-        Ticket.from_vimaly(ticket_data, ticket_type_map, bin_map, custom_fields)
+      tickets = []
+      i = 0
+
+      while (i * max_results) <= 5000
+        # Without max-results the request defaults to 200.
+        # The maximum max-results is 500, so will do 10 requests to get our maximum of 5000 tickets.
+        search_response = get("/ticket-search?text=#{query_string}&state=#{state}&max-results=#{max_results}&page-token=#{i * max_results}")
+
+        response_tickets = if search_response.instance_of? Array
+          search_response.map do |ticket_data|
+            Ticket.from_vimaly(ticket_data, ticket_type_map, bin_map, custom_fields)
+          end
+        elsif search_response.instance_of? Hash # Will be a hash when only 1 ticket is returned. (most likely because max_results is 1)
+          Ticket.from_vimaly(search_response, ticket_type_map, bin_map, custom_fields)
+        end
+
+        break unless search_response # If reponse is nil then we know we've got all the data
+
+        unless search_response.instance_of? Hash # If we get a hash we should continue to search
+          break if response_tickets.size < max_results # Can finish search because we've returned all there is.
+        end
+
+        tickets << response_tickets
```

Comment:
`response_tickets` is an Array and `tickets` is an Array, right? Then `<<` doesn't do what you think it does (adding the content of one to the other), it adds the `response_tickets` as one item into `tickets`. You can do this instead:

```ruby
ticket += response_tickets
```